ds/dataset.go

```diff
@@ -295,58 +294,77 @@ func (d *Dataset) SetBody(thread *starlark.Thread, _ *starlark.Builtin, args sta
 		return starlark.None, err
 	}
 
-	df := dataFormat.GoString()
-	if df == "" {
-		// default to json
-		df = "json"
+	if err := d.checkField("structure"); err != nil {
+		err = fmt.Errorf("cannot use a transform to set the body of a dataset and manually adjust structure at the same time")
+		return starlark.None, err
 	}
 
-	if _, err := dataset.ParseDataFormatString(df); err != nil {
-		return starlark.None, fmt.Errorf("invalid data_format: '%s'", df)
-	}
+	df := parseAs.GoString()
+	if df != "" {
+		if _, err := dataset.ParseDataFormatString(df); err != nil {
+			return starlark.None, fmt.Errorf("invalid parse_as format: '%s'", df)
+		}
 
-	if raw {
-		if str, ok := data.(starlark.String); ok {
-			d.write.SetBodyFile(qfs.NewMemfileBytes(fmt.Sprintf("data.%s", df), []byte(string(str))))
-			d.modBody = true
-			d.bodyCache = nil
-			return starlark.None, nil
+		str, ok := data.(starlark.String)
+		if !ok {
+			return starlark.None, fmt.Errorf("expected data for '%s' format to be a string", df)
 		}
 
-		return starlark.None, fmt.Errorf("expected raw data for body to be a string")
+		d.write.SetBodyFile(qfs.NewMemfileBytes(fmt.Sprintf("data.%s", df), []byte(string(str))))
```

Comment:
Use "body.%s" instead of "data.%s".

Reply:
totally. makes me want this more: https://github.com/qri-io/dataset/issues/188